vrp/client/veh_blacklist.lua

```diff
@@ -0,0 +1,64 @@
+if not vRP.modules.veh_blacklist then return end
+
+local VehBlacklist = class("VehBlacklist", vRP.Extension)
+
+-- METHODS
+
+function VehBlacklist:__construct()
+  vRP.Extension.__construct(self)
+
+  self.veh_models = {} -- map of model hash
+  self.interval = 10000
+
+  -- task: remove vehicles
+  Citizen.CreateThread(function()
+    while true do 
+      Citizen.Wait(self.interval)
+
+      local vehicles = {}
+
+      local it, veh = FindFirstVehicle()
+      if veh then table.insert(vehicles, veh) end
+
+      while true do
+        local ok, veh = FindNextVehicle(it)
+        if ok and veh then 
+          table.insert(vehicles, veh) 
+        else
+          EndFindVehicle(it)
+          break
+        end
+      end
+
+      for _, veh in ipairs(vehicles) do
+        if self.veh_models[GetEntityModel(veh)] then
+          local cid, model = vRP.EXT.Garage:getVehicleInfo(veh)
+          if not cid and vRP.EXT.Base.cid ~= cid then
```

Comment:
I'm not sure about the intent of this condition, `not cid` means that `cid` is nil, so `vRP.EXT.Base.cid ~= cid` is equivalent to `vRP.EXT.Base.cid ~= nil`. I think `not cid` is enough, I don't know what will happen if it tries to delete other players' vehicles.

Reply:
I did run some tests with 2 players and it worked fine. Tried different vehicles of different jobs/groups that is in blacklist.
But for safety, I changed the line to not cid only.

Reply:
It's because `vRP.EXT.Base.cid ~= nil` will return true most of the time.